src/wiki/plugins/linknetwork/views.py

```diff
@@ -0,0 +1,127 @@
+    paginate_by = 50
+    model = models.InternalLink
+
+    @method_decorator(get_article(can_read=True))
+    def dispatch(self, request, article, *args, **kwargs):
+        # If the article has been deleted, show a special page.
+        if article.current_revision and article.current_revision.deleted:
+            return redirect("wiki:deleted", article_id=article.id)
+
+        if not article.can_read(request.user):
+            return response_forbidden(request, article, read_denied=True)
+
+        return super().dispatch(request, article, *args, **kwargs)
+
+    def get_queryset(self):
+        # TODO: This filter should use query logic, instead of python filtering, as much as possible.
+        return [
+            link
+            for link in self.model.objects.filter(to_article=self.article).all()
+            if link.from_article.can_read(self.request.user)
+        ]
+
+    def get_context_data(self, **kwargs):
+        # Apparently a standard hack for (ListView, ArticleMixin) classes
+        kwargs_article = ArticleMixin.get_context_data(self, **kwargs)
+        kwargs_listview = ListView.get_context_data(self, **kwargs)
+        kwargs.update(kwargs_article)
+        kwargs.update(kwargs_listview)
+        return kwargs
+
+
+class LinkNetwork(ListView, ArticleMixin):
+    template_name = "wiki/plugins/linknetwork/linknetwork.html"
+    allow_empty = True
+    context_object_name = "linknetwork"
+    paginator_class = WikiPaginator
+    paginate_by = 50
+    model = models.InternalLink
+
+    @method_decorator(get_article(can_read=True))
+    def dispatch(self, request, article, *args, **kwargs):
+        if not article.can_read(request.user):
+            return response_forbidden(request, article, read_denied=True)
+        return super().dispatch(request, article, *args, **kwargs)
+
+    def get_queryset(self):
+        self.nodes = [
+            article
+            for article in wiki_models.Article.objects.all()
+            if article.get_absolute_url().startswith(self.article.get_absolute_url())
+            if article.can_read(self.request.user)
+        ]
+        # For network display of the namespace, it would be nice to also pass
+        # the isolated nodes, with no outgoing and no incoming edges. That
+        # would probably also require de-activating the paginator.
+        return self.model.objects.filter(
+            Q(from_article__in=self.nodes)
+            & (
+                Q(to_article__in=self.nodes)
+                | Q(to_nonexistant_url__startswith=self.article.get_absolute_url())
+            )
+        )
+
+    def get_context_data(self, **kwargs):
+        # Apparently a standard hack for (ListView, ArticleMixin) classes
+        kwargs_article = ArticleMixin.get_context_data(self, **kwargs)
+        kwargs_listview = ListView.get_context_data(self, **kwargs)
+        kwargs.update(kwargs_article)
+        kwargs.update(kwargs_listview)
+        return kwargs
+
+
+class GlobalUpdate(View):
```

Comment:
What does this view do? It doesn't seem to have any authentication protection or permission checking?